packages/core/src/util/file-type.ts

```diff
@@ -85,20 +85,41 @@ export function isLikelyTextFile(fileBuffer: Buffer): boolean {
   return readableChars / totalChars > threshold;
 }
 
+export type DetectedFileType = {
+  mimeType: string;
+  fileExtension: string;
+};
+
 /**
  * The function `detectFileType` uses magic numbers to determine the file type of a given file.
  * Magic numbers are unique sequences of bytes that identify the file format or protocol.
+ *
+ * In this case, the function reads the first 5 bytes (the magic number) of the file buffer and
+ * matches it against known file type headers.
+ *
+ * @param document - A string that represents the contents of a file. It will be converted to a
+ * Buffer in order to be processed. Prefer `detectFileTypeBuf` if you already have a Buffer object.
+ * @returns a string representing the detected file type.
+ */
+export function detectFileType(document: string): DetectedFileType {
```

Comment:
by separating this logic of slicing the buffer into detectFileType, we know have different behavior under the hood when we call with detectFileType and detectFileTypeBuf, since in one version it gets all the buffer and the other only 6 bytes. This is the root cause of the bug identified below. I would suggest having a single detectFileType function that takes a buffer like in my reverted PR https://github.com/metriport/metriport/pull/1884/files#r1582280276 and convert things to a buffer where relevant before calling it.

Reply:
Sounds good, I'll update it